src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
+        if ((size - 1) > index) {
+            for (int i = 0; i < size; i++) {
+                if (i == index) {
+                    oldValue = node.value;
+                    node.value = value;
+                    break;
+                }
+                node = node.next;
+            }
+            return oldValue;
+        } else {
+            throw new IndexOutOfBoundsException("size; " + size + "index: " + index);
+        }
+
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = head;
+        if (index > 0 && index < (size)) {
+            for (int i = 0; i < size; i++) {
+                if (i == index) {
+                    if (node.next != null) {
+                        node.next.prev = node.prev;
+                    }
+                    node.prev.next = node.next;
+                    size--;
+                    return node.value;
+                } else {
+                    node = node.next;
+                }
+            }
+        } else if (index == 0) {
+            if (size != 1) {
+                node.next.prev = null;
+                head = node.next;
+                size--;
+                return node.value;
+            } else {
+                head = null;
+                tail = null;
+                size--;
+                return node.value;
+            }
+
+        }
+        {
+            throw new IndexOutOfBoundsException("size; " + size + "index: " + index);
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        if (size == 1) {
+            head = null;
+            size--;
+            return true;
+        } else {
+            for (int i = 0; i < size; i++) {
+                if (node.value == null && object == null) {
+                    node.next.prev = node.prev;
+                    node.prev.next = node.next;
+                    size--;
+                    return true;
+                } else if (node.value != null) {
+                    if (node.value.equals(object)) {
+                        if (node.equals(tail)) {
+                            node.prev.next = null;
+                            tail = node.prev;
+                            size--;
+                            return true;
+                        } else if (node.equals(head)) {
+                            head = node.next;
+                            head.prev = null;
+                            size--;
+                            return true;
+                        } else {
+                            node.prev.next = node.next;
+                            node.next.prev = node.prev;
+                            size--;
+                            return true;
+                        }
+                    }
+                    node = node.next;
+                } else {
+                    node = node.next;
+                }
+            }
+        }
         return false;
     }
```

Comment:
In the `remove(T object)` method, you should also use the `unlink(Node node)` method to remove all the links when you remove a Node.